sql/registry.go

```diff
@@ -0,0 +1,73 @@
+package sql
+
+import (
+	"database/sql"
+	"fmt"
+	"sync"
+)
+
+// DriverRegistry is the interface for Driver registries.
+type DriverRegistry interface {
+	Has(name string) bool
+	Add(name string, driver *Driver) error
+	Get(name string) (*Driver, error)
+}
+
+// DefaultDriverRegistry is the default DriverRegistry implementation.
+type DefaultDriverRegistry struct {
+	drivers map[string]*Driver
+	mutex   sync.RWMutex
+}
+
+// NewDefaultDriverRegistry returns a new DefaultDriverRegistry.
+func NewDefaultDriverRegistry() *DefaultDriverRegistry {
+	return &DefaultDriverRegistry{
+		drivers: make(map[string]*Driver),
+	}
+}
+
+// Add adds and register a given Driver for a name.
+func (r *DefaultDriverRegistry) Add(name string, driver *Driver) (err error) {
+	if r.Has(name) {
+		return err
```

Comment:
If it returns `nil` for already registered drivers on purpose, it would be less confusing if it returns `nil` explicitly.
```suggestion
		return nil
```

Reply:
agreed :+1:  done